libs/deal-ticket/src/hooks/use-max-size.ts

```diff
@@ -66,10 +66,12 @@ export const useMaxSize = ({
         .div(marginFactor)
         .div(toBigNum(price, decimalPlaces));
     } else {
+      const effectivePrice =
+        type === OrderType.TYPE_LIMIT ? marketPrice || price : marketPrice;
```

Comment:
Something seems wrong here. The price value passed into the hook is already derived using order type in getDerivedPrice?

Reply:
So I think just using `price` is correct here, and there is no need to check order type

Reply:
In cross mode orders mark price is used to calculate required margin for both limit and market orders. If market is in auction there is no market price and estimate is calculated using price from deal ticket.

Use always marketPrice until order type is limit and there is no marketPrice.